git_submodules/lib/dependabot/git_submodules/update_checker.rb

```diff
@@ -10,20 +12,30 @@
 module Dependabot
   module GitSubmodules
     class UpdateChecker < Dependabot::UpdateCheckers::Base
+      extend T::Sig
+
+      sig { override.returns(T.nilable(T.any(String, Gem::Version))) }
       def latest_version
-        @latest_version ||= fetch_latest_version
+        @latest_version ||=
+          T.let(
+            fetch_latest_version,
+            T.nilable(T.any(String, Gem::Version))
+          )
       end
 
+      sig { override.returns(T.nilable(T.any(String, Gem::Version))) }
       def latest_resolvable_version
         # Resolvability isn't an issue for submodules.
         latest_version
       end
 
+      sig { override.returns(T.nilable(T.any(String, Gem::Version))) }
```

Comment:
Should we be using `Dependabot::Version`? 🤔 

```suggestion
      sig { override.returns(T.nilable(T.any(String, Dependabot::Version))) }
```

Reply:
Ideally, yes (Actually, I think it should be a generic that accepts `Dependabot::ECOSYSTEM::Version`). But `Gem::Version` and `Dependabot::Version` are used interchangeably throughout the codebase at the moment. Once it's fully typed it will be much easier to refactor.